cloudinit/net/dhcp.py

```diff
@@ -288,17 +311,22 @@ def parse_static_routes(rfc3442):
               first parsing error.
 
     E.g.
-    sr = parse_state_routes("32,169,254,169,254,130,56,248,255,0,130,56,240,1")
-    sr = [
+    sr=parse_static_routes("32,169,254,169,254,130,56,248,255,0,130,56,240,1")
+    sr=[
         ("169.254.169.254/32", "130.56.248.255"), ("0.0.0.0/0", "130.56.240.1")
     ]
 
+    sr2 = parse_static_routes("24.191.168.128 192.168.128.1,0 192.168.128.1")
+    sr2 = [
+        ("191.168.128.0/24", "192.168.128.1"), ("0.0.0.0/0", "192.168.128.1")
+    ]
+
     Python version of isc-dhclient's hooks:
        /etc/dhcp/dhclient-exit-hooks.d/rfc3442-classless-routes
     """
     # raw strings from dhcp lease may end in semi-colon
     rfc3442 = rfc3442.rstrip(";")
-    tokens = rfc3442.split(',')
+    tokens = re.split(r"[, .]", rfc3442)
```

Comment:
Let's add something in the comments to indicate that we're parsing two formats of rfc3442,  one from dhcpcd and one from dhclient (isc). 

I found this comment in NetworkManager which has a few more formats we may add later:

https://github.com/heftig/NetworkManager/blob/master/src/dhcp-manager/nm-dhcp-client.c#L978

Reply:
I'm not sure I get you meant as there was multiple possibility ; 
should it be with the @param in the comment to allow people to understand it when they use the fonction?
Should it be just some new lines specifying that this function need to support multiple format ?
Should it be on line 329 on which we already have code on.
Is it sufficient to be in the commit message?

Reply:
I did what I thought made the most sense.  Please let me know if this wasn't what you had in mind.